common/src/types/v0/store/node.rs

```diff
@@ -1,31 +1,104 @@
 //! Definition of node types that can be saved to the persistent store.
 
-use crate::types::v0::{
-    openapi::models,
-    store::{
-        definitions::{ObjectKey, StorableObject, StorableObjectType},
-        ResourceUuid,
+use crate::{
+    types::v0::{
+        openapi::models,
+        store::{
+            definitions::{ObjectKey, StorableObject, StorableObjectType},
+            ResourceUuid,
+        },
+        transport::{self, NodeId},
     },
-    transport::{self, NodeId},
+    IntoOption,
 };
 use serde::{Deserialize, Serialize};
 use std::collections::HashMap;
 
+/// Generic labels associated with the node
 pub type NodeLabels = HashMap<String, String>;
 
+/// Data relating to the cordoning of a node
+#[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
+pub struct CordonedState {
+    /// labels used to enforce a node cordon
+    pub cordonlabels: Vec<String>,
```

Comment:
nit: cordon`_`labels ?

Reply:
I'm trying to avoid the hyphens vs underscores issue in identifiers/paths by using neither unless necessary.
I think here cordonlabels is legible enough not to need it IMO.

Reply:
I have a slight preference for the "_" :) but it's fine with me as it is :+1: 
q: what about removing "cordon" altogether since we're in the CordonedState?
So just "labels", even simpler.

Reply:
You could, but if you call them just "labels" it's less obvious that they become (or are copied from) the same set of labels as the "cordonlabels" in the DrainState when the state changes. That would make harder for developers to follow IMO.